src/progress_bar.rs

```diff
@@ -615,6 +635,26 @@ impl MultiProgress {
         self.join_impl(true)
     }
 
+    /// Modify the state on inserting/adding a progress bar
+    fn extend_state(state: &mut MultiProgressState) -> usize {
+        let object_idx;
+        if let Some(i) = state.free_set.iter().next().cloned() {
+            state.objects[i] = Some(MultiObject {
+                done: false,
+                draw_state: None,
+            });
+            state.free_set.remove(&i);
```

Comment:
Oops, this operation is unsupported for heaps.

Reply:
Hmm, I think `BTreeSet` could also be a decent option here, what do you think?

Reply:
To reduce memory footprint? Ok, let's try.